app/models/user.rb

```diff
@@ -4,4 +4,5 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   has_many :comments
+  belongs_to :cart
```

Comment:
I don't think it makes sense for the user to belong to the cart.  This should be user has_one :cart . Else you would not be able to create a new user without a cart id -- it wouldn't be valid.

Reply:
Updated.